### PR TITLE
Docs: Put more structure to Architecture page

### PR DESCRIPTION
**Pull request by gziolo**
## What?

I got inspired by the input from @Mamaduka about the current structure of [Architecture page](https://developer.wordpress.org/block-editor/explanations/architecture/) in the block editor handbook. He mentioned many concepts on the page are missing today on this page. I started looking first how to better structure the document.

The follow-up task would be to discuss what topics we should include on this page.

### CODE REVIEW

File: docs/explanations/architecture/README.md [hunk 1]
```diff
@@ -2,12 +2,16 @@
...
+## Gutenberg Repository
+
 -   [Modularity and WordPress Packages](/docs/explanations/architecture/modularity.md).
--   [Block Editor Performance](/docs/explanations/architecture/performance.md).
--   What are the decision decisions behind the Data Module?
--   [Why is Puppeteer the tool of choice for end-to-end tests?](/docs/explanations/architecture/automated-testing.md)
--   [What's the difference between the different editor packages? What's the purpose of each package?](/docs/explanations/architecture/modularity.md#whats-the-difference-between-the-different-editor-packages-whats-the-purpose-of-each-package)
--   [Template and template parts flows](/docs/explanations/architecture/full-site-editing-templates.md)
+-   [Understand the repository folder structure](/docs/contributors/folder-structure.md).
+-   **Outdated!** [Why is Puppeteer the tool of choice for end-to-end tests?](/docs/explanations/architecture/automated-testing.md).
```
Comment (priethor) [inline docs/explanations/architecture/README.md:17]: Although not ideal, marking the section as outdated is a good quick compromise and better than leaving it as up-to-date or than completely removing the page, as it still provides context and many tests are still created in Puppeteer.

Updating `/docs/explanations/architecture/automated-testing.md` would be a good next step for a follow-up PR cc @kevin940726  @talldan

File: docs/explanations/architecture/styles.md [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
 2. [Block styles](#block-styles)
-  - [From UI controls to HTML markup](#from-ui-controls-to-html-markup)
-  - [Block Supports API](#block-supports-api)
-  - [Current limitations of the Block Supports API](#current-limitations-of-the-block-supports-api)
+
+-   [From UI controls to HTML markup](#from-ui-controls-to-html-markup)
+-   [Block Supports API](#block-supports-api)
+-   [Current limitations of the Block Supports API](#current-limitations-of-the-block-supports-api)
+
 3. [Global styles](#global-styles)
```
Comment (priethor) [inline docs/explanations/architecture/styles.md:12]: Although we are trying to remove the `Global` from the `Global Styles` high-level concept and UI, this page dives more into implementation details, where the `global` is still valid and important to follow.

@oandregal would you mind updating this file as you see fit when you have some time?

**Review by priethor — APPROVED** [review]
LGTM as a quick iteration!

File: docs/explanations/architecture/README.md (hunk 1 above)
Comment (kevin940726) [reply docs/explanations/architecture/README.md:17]: I somehow totally forgot that this exists. Thanks for the ping! I'll see what I can do.

File: docs/explanations/architecture/styles.md (hunk 2 above)
Comment (oandregal) [reply docs/explanations/architecture/styles.md:12]: Sure, happy to. I'd need some direction as to how to update it.

My understanding is that the difference between block-level styles (serialized to post content as part of the block) and global-level styles (part of a stylesheet that's generated dynamically) is still relevant.

Comment (priethor) [reply docs/explanations/architecture/styles.md:12]: If you think it's still fully up to date, there is no need to update it until we land the styles persistence across themes in #47420 👍

**Review by aristath — APPROVED** [review]

**Comment by gziolo** [comment]
I opened a follow-up discussion https://github.com/WordPress/gutenberg/discussions/49442, so we figure out what's missing on this page.

**Pull request merged**